shared/src/commonMain/kotlin/io.newm.shared/repositories/CardanoWalletRepository.kt

```diff
@@ -50,38 +46,47 @@ data class File(
     val src: String
 )
 
+fun List<LedgerAssetMetadata>.getTrackFromMusicMetadataV1(): NFTTrack? {
+    val imageId = this.find { it.key == "image" }?.value ?: return null
+    val artistNames = this.find { it.key == "artists" }
+        ?.children
+        ?.flatMap { artist ->
+            artist.children.mapNotNull { detail ->
+                if (detail.key == "name") detail.value as? String else null
+            }
+        }.orEmpty()
+
+    val songSrcId = this.find { it.key == "image" }?.value ?: return null
+
+    return NFTTrack(
+        songName = find { it.key == "song_title" }?.value ?: "Where is the song name?",
```

Comment:
Doesn't this mean "Where is the song name?" will be shown to users?  If it's not a required field then it should be optional.  But I would expect it to be required, and if it's missing then the return value should be nil, or throw an error.  I would expect it to be required and throw an error, if it's not possible to mint a song with no title.